cdc/log.cc

```diff
@@ -657,13 +658,18 @@ class transformer final {
 
         query::column_id_vector static_columns, regular_columns;
 
-        auto sk = column_kind::static_column;
-        auto rk = column_kind::regular_column;
         // TODO: this assumes all mutations touch the same set of columns. This might not be true, and we may need to do more horrible set operation here.
-        for (auto& [r, cids, kind] : { std::tie(p.static_row().get(), static_columns, sk), std::tie(p.clustered_rows().begin()->row().cells(), regular_columns, rk) }) {
-            r.for_each_cell([&](column_id id, const atomic_cell_or_collection&) {
-                auto& cdef =_schema->column_at(kind, id);
-                cids.emplace_back(id);
+        if (!p.static_row().empty()) {
+            p.static_row().get().for_each_cell([&] (column_id id, const atomic_cell_or_collection&) {
+                auto& cdef =_schema->column_at(column_kind::static_column, id);
+                static_columns.emplace_back(id);
+                columns.emplace_back(&cdef);
+            });
+        }
+        if (!p.clustered_rows().empty()) {
+            p.clustered_rows().begin()->row().cells().for_each_cell([&] (column_id id, const atomic_cell_or_collection&) {
+                auto& cdef =_schema->column_at(column_kind::regular_column, id);
+                regular_columns.emplace_back(id);
```

Comment:
Could just have added an `.. cids, kind, include] : { std::tie(..., !p.static_row().empty()), std::tie(..., !p.clustered_rows().empty()))` and avoided the repetition. Remember: if it is worth doing, it is worth doing as a loop.

Reply:
https://www.youtube.com/watch?v=WWaLxFIVX1s

Reply:
Not sure what that is supposed to entail? Would you care to defend your code repetition more verbosely?

Reply:
Sure.

>  if it is worth doing, it is worth doing as a loop.

I strongly disagree. The loop that was already there was unreadable, and you're proposing to make it even more unreadable.

I agree that code repetition is bad in general, but sometimes it's better than the alternatives.